sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -337,7 +337,9 @@ def _download_blob_options(
     :returns: A dictionary containing the download blob options.
     :rtype: Dict[str, Any]
     """
-    if length is not None:
+    if length is not None and offset is None:
+        raise ValueError("Offset must be provided if length is provided.")
+    if length:
```

Comment:
Hmmm I didn't realize the line underneath this also needed the length None check... and I don't love this double if now. Maybe put this back to how you had it? (Sorry!)

Reply:
Yeah it's a little silly when I was looking back it it yesterday.. but it is what it is 😄